packages/client-core/src/admin/components/Project/UserPermissionDrawer.tsx

```diff
@@ -78,8 +78,12 @@ const UserPermissionDrawer = ({ open, project, onClose }: Props) => {
     }
 
     try {
-      await ProjectService.createPermission(userInviteCode, project.id)
-      await ProjectService.fetchProjects()
+      const newPermission = await ProjectService.createPermission(userInviteCode, project.id)
+      if (
+        Array.isArray(project?.projectPermissions) &&
+        project.projectPermissions.findIndex((permission) => permission.id === newPermission.id) < 0
+      )
+        project.projectPermissions = project.projectPermissions.concat([newPermission])
```

Comment:
`project` here is coming from a reactive store (inside useFind), it will cause desync issues inside hookstate. instead, we should make a `projectPermissions` state inside this react component that is set here, thus can be reacted to